core/async_job/queue.ex

```diff
@@ -125,28 +118,70 @@ defmodule AntikytheraCore.AsyncJob.Queue do
+    {j1, t, :running} = Map.fetch!(jobs, job_id)
+    case j1.remaining_attempts do
+      1 ->
+        abandon_job(q, target_job_info, j1) |> requeue_if_recurring(j1, job_id, now_millis)
+      remaining ->
+        attempt_job(q, target_job_info, j1, t, remaining, index_waiting)
+    end
+    |> move_jobs_running_too_long(threshold_time)
+  end
+
+  defp abandon_job(%__MODULE__{jobs:           jobs,
+                               abandoned_jobs: abandoned_jobs} = q,
+                   {{_, job_id}, index_running2} = _target_job_info,
+                   job) do
+    jobs2 = Map.delete(jobs, job_id)
+    abandoned_jobs2 = [{job_id, job} | abandoned_jobs]
+    %__MODULE__{q | jobs: jobs2, index_running: index_running2, abandoned_jobs: abandoned_jobs2}
+  end
+
+  defp attempt_job(%__MODULE__{jobs: jobs} = q,
+                   {{_, job_id} = job_key, index_running2} = _target_job_info,
+                   job,
+                   time,
+                   remaining,
+                   index_waiting) do
+    job2 = %AsyncJob{job | remaining_attempts: remaining - 1}
+    jobs2 = Map.put(jobs, job_id, {job2, time, :waiting})
+    index_waiting2 = :gb_sets.add(job_key, index_waiting)
+    %__MODULE__{q | jobs: jobs2, index_waiting: index_waiting2, index_running: index_running2}
+  end
+
   defunp move_now_runnable_jobs(q :: v[t], now_millis :: v[pos_integer]) :: t do
     %__MODULE__{q | brokers_to_notify: []}
     |> move_now_runnable_jobs_impl(now_millis)
   end
-  defp move_now_runnable_jobs_impl(%__MODULE__{jobs:              jobs,
-                                               index_waiting:     index_waiting,
-                                               index_runnable:    index_runnable,
-                                               brokers_waiting:   brokers_waiting,
-                                               brokers_to_notify: brokers_to_notify} = q,
+  defp move_now_runnable_jobs_impl(%__MODULE__{index_waiting: index_waiting} = q,
                                    now_millis) do
-    case take_smallest_with_earlier_timestamp(index_waiting, now_millis) do
-      nil                                     -> q
-      {{_, job_id} = job_key, index_waiting2} ->
-        jobs2           = Map.update!(jobs, job_id, fn {j, t, :waiting} -> {j, t, :runnable} end)
-        index_runnable2 = :gb_sets.add(job_key, index_runnable)
-        q2              = %__MODULE__{q | jobs: jobs2, index_waiting: index_waiting2, index_runnable: index_runnable2}
-        case brokers_waiting do
-          []       -> q2
-          [b | bs] -> %__MODULE__{q2 | brokers_waiting: bs, brokers_to_notify: [b | brokers_to_notify]}
-        end
-        |> move_now_runnable_jobs_impl(now_millis)
-    end
+    target_job_info = take_smallest_with_earlier_timestamp(index_waiting, now_millis)
+    move_now_runnable_job(q, target_job_info, now_millis)
+  end
+
+  defp move_now_runnable_job(q, nil = _target_job_info, _now_millis), do: q
```

Comment:
We can infer the role of this argument from the argument name of the same function.
```suggestion
  defp move_now_runnable_job(q, nil, _now_millis), do: q
```

Reply:
I fixed it https://github.com/access-company/antikythera/pull/51/commits/43b95be6693fa8736bf6caf8bd391781a4ccc141